### PR TITLE
adding page about version compatibility

### PR DESCRIPTION
**Pull request by samuelcolvin**
definitely more things we could add here, but I think this is a good start.

Selected reviewer: @tpdorsey

### CODE REVIEW

**Comment by samuelcolvin** [comment]
please review.

File: docs/version-compatibility.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Methods marked as `deprecated` will not be removed until the next major release, V3.
+
+Of course some apparently safe changes and bug fixes will inevitably break some users' code - obligatory link to [XKCD](https://m.xkcd.com/1172/).
+
+The following changes will **NOT** be considered breaking changes, and may occur in minor releases:
+
+* changing the format of `ref` as used in JSON Schema
+* changing the `message` and `context` fields of `ValidationError` errors, `type` will not change - if you're programmatically parsing error messages, you should use `type`
+* adding new keys to `ValidationError` errors - e.g. we intend to add `line_number` and `column_number` to errors when validating JSON once we migrate to a new JSON parser
```
Comment (tpdorsey) [inline docs/version-compatibility.md:22]:
```suggestion
* Adding new keys to `ValidationError` errors &mdash; e.g. we intend to add `line_number` and `column_number` to errors when validating JSON once we migrate to a new JSON parser.
```

File: docs/version-compatibility.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+* adding new `ValidationError` errors
```
Comment (tpdorsey) [inline docs/version-compatibility.md:23]:
```suggestion
* Adding new `ValidationError` errors.
```

File: docs/version-compatibility.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+* changing `repr` even of public classes
```
Comment (tpdorsey) [inline docs/version-compatibility.md:24]:
```suggestion
* Changing `repr` even of public classes.
```

File: docs/version-compatibility.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+## V3 and Beyond
```
Comment (tpdorsey) [inline docs/version-compatibility.md:26]:
```suggestion
## Pydantic V3 and beyond
```

File: docs/version-compatibility.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+We expect to make new major releases roughly once a year going forward, although as mentioned above, any associated breaking changes should be trivial to fix compared to the V1 to V2 transition.
```
Comment (tpdorsey) [inline docs/version-compatibility.md:28]:
```suggestion
We expect to make new major releases roughly once a year going forward, although as mentioned above, any associated breaking changes should be trivial to fix compared to the V1-to-V2 transition.
```

File: mkdocs.yml [hunk 6]
```diff
@@ -47,6 +47,7 @@ nav:
   - Welcome to Pydantic: index.md
   - Installation: install.md
   - Migration Guide: migration.md
+  - version-compatibility.md
```
Comment (tpdorsey) [inline mkdocs.yml:50]:
```suggestion
  - Version Compatibility: version-compatibility.md
```

File: docs/version-compatibility.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+First of all, we recognize that the transitions from Pydantic V1 to V2 has been and will be painful for many users.
```
Comment (tpdorsey) [inline docs/version-compatibility.md:1]:
```suggestion
First of all, we recognize that the transitions from Pydantic V1 to V2 has been and will be painful for some users.
```

**Review by tpdorsey — APPROVED** [review]

File: docs/version-compatibility.md [hunk 8]
```diff
@@ -1,28 +1,28 @@
-First of all, we recognize that the transitions from Pydantic V1 to V2 has been and will be painful for many users.
-We're sorry about this pain :pray:, it was an unfortunate but unavoidable consequence of both the rewrite to use Rust and of correcting some design mistakes of V1.
+First of all, we recognize that the transitions from Pydantic V1 to V2 has been and will be painful for some users.
+We're sorry about this pain :pray:, it was an unfortunate, but unavoidable, consequence of both the rewrite to use Rust and of correcting some design mistakes of V1.
```
Comment (adriangb) [inline docs/version-compatibility.md:2]:
```suggestion
We're sorry about this pain :pray:, it was an unfortunate but necessary step to correct design mistakes of V1.
```

I'd just focus on the design mistakes of V1. I think that's where most of the breaking changes are coming from anyway.

**Review by adriangb — APPROVED** [review]

**Pull request merged**